### PR TITLE
[CNFT2-1410] Search Target Question

### PR DESCRIPTION
**Pull request by lramakr**
## Description

[CNFT2-1410] - Search Target Question UI implementation

## Tickets

* [Jira Ticket]https://cdc-nbs.atlassian.net/browse/CNFT2-1410

## Steps to verify

1. Navigate to manage/pages
2. Select any page name that navigate to edit the particular page
3. Click on pencil Icon against any row items
4. That provide option to add Logic
5. There is a Search Target Question button. Click on it
6. This take you to Target question Model
7. There you can expand the tree view and select / unselect / select all option
8. Click continue button bring you back to Logic page with selected option values

[CNFT2-1410]: https://cdc-nbs.atlassian.net/browse/CNFT2-1410?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mpeels** [comment]
This includes code from your other pull requests. Any way we can isolate the commits for the given story to reduce the scope of changes?

File: apps/modernization-ui/src/apps/page-builder/pages/ValuesetLibrary/ValuesetLibraryTable.tsx [hunk 1]
```diff
@@ -284,20 +295,20 @@ export const ValuesetLibraryTable = ({ summaries, labModalRef, pages }: Props) =
...
+                    pageSize={pages?.pageSize}
+                    totalResults={pages?.totalElements}
+                    currentPage={pages?.currentPage}
                     handleNext={setCurrentPage}
                     sortData={handleSort}
                     selectable
                     handleSelected={handleSelected}
                     rangeSelector={true}
                     isLoading={isLoading}
+                    context={useContext(ValueSetsContext)}
```
Comment (achapm) [inline apps/modernization-ui/src/apps/page-builder/pages/ValuesetLibrary/ValuesetLibraryTable.tsx:311]: We should use createContext and useContext and not need to pass through props:
https://react.dev/learn/passing-data-deeply-with-context

**Review by achapm — CHANGES_REQUESTED** [review]

File: apps/modernization-ui/src/apps/page-builder/pages/QuestionLibrary/QuestionLibraryTable.tsx [hunk 2]
```diff
@@ -225,6 +235,7 @@ export const QuestionLibraryTable = ({ summaries, pages, qtnModalRef }: Props) =
                     handleSelected={handleSelected}
                     rangeSelector={true}
                     isLoading={isLoading}
+                    context={useContext(QuestionsContext)}
```
Comment (achapm) [inline apps/modernization-ui/src/apps/page-builder/pages/QuestionLibrary/QuestionLibraryTable.tsx:238]: We should use createContext and useContext and not need to pass through props:
https://react.dev/learn/passing-data-deeply-with-context

**Review by achapm — CHANGES_REQUESTED** [review]
Please merge with main branch and resolve any conflicts.

**Comment by lramakr** [comment]
Hi @achapm , @mpeels , 
    The Squash and merge button is not enabled. So I dont see the option to merge. Kindly check

**Comment by mpeels** [comment]
Closed as this functionality will be split into its own separate story and implementation.

**Pull request closed without merging**